rclpy/src/rclpy/names.hpp

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__NAMES_HPP_
+#define RCLPY__NAMES_HPP_
+
+#include <pybind11/pybind11.h>
+
+#include <string>
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+/// Validate a topic name and return error message and index of invalidation.
+/**
+ * Does not have to be a fully qualified topic name.
+ * The topic name is not expanded.
+ *
+ * Raises MemoryError if memory could not be allocated
+ * Raises RCLError if an unexpected error happened while validating the topic name
+ *
+ * \param[in] topic_name name of the topic to be validated
+ * \return tuple of error message and invalid index if invalid, or
+ * \return None if valid
+ */
+py::object
+get_validation_error_for_topic_name(const char * topic_name);
+
+/// Validate a full topic name and return error message and index of invalidation.
+/**
+ * Must be a fully qualified topic name.
+ *
+ * Raises MemoryError if memory could not be allocated
+ * Raises RCLError if an unexpected error happened while validating the topic name
+ *
+ * \param[in] topic_name name of the topic to be validated
+ * \return tuple of error message and invalid index if invalid, or
+ * \return None if valid
+ */
+py::object
+get_validation_error_for_full_topic_name(const char * topic_name);
+
+/// Validate a namespace and return error message and index of invalidation.
+/**
+ * Raises MemoryError if memory could not be allocated
+ * Raises RCLError if an unexpected error happened while validating the namespace
+ *
+ * \param[in] namespace namespace to be validated
+ * \return tuple of error message and invalid index if invalid, or
+ * \return None if valid
+ */
+py::object
+get_validation_error_for_namespace(const char * namespace_);
+
+/// Validate a node name and return error message and index of invalidation.
+/**
+ * Raises MemoryError if memory could not be allocated
+ * Raises RuntimeError if an unexpected error happened while validating the node name
+ *
+ * \param[in] node_name name of the node to be validated
+ * \return tuple of error message and invalid index if invalid, or
+ * \return None if valid
+ */
+py::object
+get_validation_error_for_node_name(const char * node_name);
+
+/// Expand a topic name
+/**
+ * Raises ValueError if the topic name, node name, or namespace are not valid.
```

Comment:
These last three functions can raise RCLError. Is that omitted intentionally?

Reply:
Added note about raising RCLError in c269b6ac92dd3f15aafebffd3117055cfc34cad2